gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_widgets.dart

```diff
@@ -78,6 +79,10 @@ class _ProTokenInputFieldState extends State<ProTokenInputField> {
           children: [
             Expanded(
               child: TextField(
+                inputFormatters: [
+                  // This ignores all sorts of (Unicode) whitespaces (not only at the ends).
+                  FilteringTextInputFormatter.deny(RegExp(r'\s')),
```

Comment:
I'm no dart coder, so forgive my ignorance. Why not use an allow() constructor instead such as `.allow(RegExp(r'[a-zA-Z0-9]'))`  and also limit the length of the field to the max length of a token?
(also note that I couldn't confirm that it is the correct pattern for a token but it must be available somewhere in the server code)

Reply:
Well, the proper filtering to achieve your proposal is a bit more evolving, but doable to a certain extent (there are preconditions about the certain element positions as well as the use of base58 encoding that won't fit well as input formatters). 

The downside I see is that by implementing a good input filtering scheme we might be helping non-behaved users to guess a valid token by inputting a whole lot of non-sense characters that we just strip out leaving only the valid candidates, reducing the guessing effort.

There is already a more complete validation mechanism for the structure of the pro token defined in `gui/lib/core/pro_token.dart`, which is what backs the error messages we see when it's too long, too short, "corrupted" (invalid structure) etc.

If the guessing game I mentioned above is not an issue, then we can leverage the `inputFormatters` to remove parts of the input value that would cause some error conditions to surface in the UI. We can use them to constrain the length as you suggested as well as filtering characters that we can be sure are invalid (such as `I`, `O`, `0`, etc).